src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -262,7 +271,12 @@ protected WebResourceResponse interceptRequest(String rawUrl) {
         byte[] blob = this.dcAppMsg.getWebxdcBlob(path);
         if (blob == null) {
           if (internetAccess) {
-            return null; // do not intercept request
+            HttpResponse httpResponse = rpc.getHttpResponse(dcContext.getAccountId(), rawUrl);
+            String mimeType = httpResponse.getMimetype();
+            if (mimeType == null) {
+              mimeType = "application/octet-stream";
+            }
+            return new WebResourceResponse(mimeType, httpResponse.getEncoding(), new ByteArrayInputStream(httpResponse.getBlob()));
```

Comment:
This is because otherwise normal traffic is redirected to the fake proxy introduced in #2539 but while this works with the webxdc-test app that loads the delta.chat page, the maps app doesn't work because OSM is not happy about the [wrong user agent](https://operations.osmfoundation.org/policies/tiles/) core is sending:
![image](https://github.com/deltachat/deltachat-android/assets/24558636/5895c17e-3c90-4b3b-9924-87df824ab858)

Ideally core should be fixed because otherwise the browsing will not respect the core proxy set like the "show full message" does, otherwise just not setting this fake proxy when internetAccess == true would be the simplest solution (ignoring the network leaks out of the configured proxy, but I suspect WebRTC also doesn't respect this proxy? otherwise that would have been used to disable it)

Reply:
cc @link2xt

Reply:
will do that, I am a bit worried about the inconsistent usage of the set sock proxy in DC, it is used only for "show full message" and it might cause false expectations we don't say anything about this traffic in Webxdc not being routed to the set proxy

Reply:
done

Reply:
There are also leaflet plugins for caching, like https://github.com/MazeMap/Leaflet.TileLayer.PouchDBCached

I think it would still make sense to use core, we could add support for defining http headers in the jsonrpc method. core respects socks5 proxy (for example tor) and on desktop we might not be able to replicate this, as dns block seem to be for the whole app there (all windows at once), or whitelisting some domains, like we currently do for mapbox.